### PR TITLE
[WIP] fix: config structure to support Kustomize 2.x

### LINKED ISSUE #129: Investigate using Kustomize 2.x
There are some breaking changes from kustomize 1.x --> 2.x that would prevent us from simply bumping the version.

ref:
- https://github.com/kubernetes-sigs/kustomize/blob/master/docs/version2.0.0.md
- https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/126#discussion_r263607147
- https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/569
- https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/611

/kind feature
/priority important-longterm
/milestone baseline
/help

### PR DESCRIPTION
**Pull request by serbrech**
**What this PR does / why we need it**:
This PR restructure the generated config to match Kustomize 2.x requirements
- use `base` to reference kustomization targets
- remove `secretGenerator commands` usage
- move generated manager patch under default
- adds `azure.env` file which contains the credentials needed. When no value is provided for the keys (as it is the case in the default file), the value is fetched from environment. This enables the following syntax : 

```shell
AZURE_CLIENT_ID=guid AZURE_CLIENT_SECRET=guid ... kustomize build ...
```

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #129

**Special notes for your reviewer**:
The change is incompatible with Kustomize 1.x as the `env` builtin source is only available from 2.x

**Please confirm that if this PR changes any image versions**
No changes to images

**Release note**:

```release-note
make use of base and support kustomize 2.x
```

### CODE REVIEW

**Comment by serbrech** [comment]
marking as WIP as it should not be merged before latest kubebuilder is used

**Comment by serbrech** [comment]
I signed it

File: config/default/azure_manager_image_patch.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: controller-manager
+  namespace: system
+spec:
+  template:
+    spec:
+      containers:
+      - image: quay.io/k8s/cluster-api-azure-controller:0.1.0-alpha.3
```
Comment (awesomenix) [inline config/default/azure_manager_image_patch.yaml]: This needs to be

```
apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: controller-manager
  namespace: system
spec:
  template:
    spec:
      containers:
      - image: ${MANAGER_IMAGE}
        imagePullPolicy: ${MANAGER_IMAGE_PULL_POLICY}
        name: manager
```

**Comment by justaugustus** [comment]
@serbrech -- please hold off on this in the short term. 
There are some updates happening in #131, which will likely change your direction. I'm trying to land that one first thing next week and then we can discuss.

**Comment by justaugustus** [comment]
@serbrech -- #131 has merged, which refactors a bit of our kustomize usage. Would you care to continue investigating support for Kustomize 2.0?

**Comment by justaugustus** [comment]
Going to close this out. I believe v1alpha2 will be using kustomize 3.x, so let's reassess once v1alpha2 lands.

**Pull request closed without merging**